src/renderer/src/components/song/context-menu/items/PlaylistChooser.tsx

```diff
@@ -1,91 +1,77 @@
-  const addToPlaylist = async (name: string) => {
-    const result = await window.api.request("playlist::add", name, props.song);
-    if (result.isError) {
-      noticeError(result.error);
-      return;
-    }
-    addNotice({
-      title: "Song added",
-      description: "Successfully added song to playlist " + name + "!",
-      variant: "success",
-      icon: <CircleCheckIcon size={20} />,
-    });
-  };
-
-  const isInPlaylist = (song: Song, playlistName: string) => {
-    if (song.playlists === undefined) {
-      return false;
-    }
-
-    return song.playlists.includes(playlistName);
-  };
-
   return (
-    <DropdownList
-      onMouseOver={() => {
-        clearTimeout(props.timeoutId());
-      }}
-      onMouseLeave={() => {
-        props.setTimeoutId(
-          setTimeout(() => {
-            props.setShowChooser(false);
-          }, 320),
-        );
-      }}
-      class="max-h-72 w-40 overflow-auto pr-1.5"
-    >
+    <div class="flex flex-col gap-1">
       <DropdownList.Item
         onClick={() => {
-          setCreatePlaylistBoxSong(props.song);
-          setShowPlaylistCreateBox(true);
           setSidebarActiveTab(SIDEBAR_PAGES.PLAYLISTS.value);
-          setPlaylistActiveScene(PLAYLIST_SCENE_LIST);
+          setPlaylistActivePage({ name: "new" });
+          props.onCreatePlaylistClick?.();
         }}
+        class="hover:bg-surface"
       >
         <span>Create Playlist</span>
-        <PlusIcon class="text-subtext" size={20} />
+        <PlusIcon size={16} />
       </DropdownList.Item>
+      <div class="h-[1px] w-full bg-surface" />
+
       <For
         fallback={<DropdownList.Item disabled={true}>No playlists...</DropdownList.Item>}
-        each={props.playlistNames}
+        each={props.playlists}
       >
-        {(child, index) => (
-          <DropdownList.Item
-            onClick={() => {
-              addToPlaylist(props.playlistNames[index()]);
-            }}
-          >
-            <span>{child}</span>
-            <Show when={isInPlaylist(props.song, props.playlistNames[index()])}>
-              <CheckIcon class="text-subtext" size={20} />
-            </Show>
-          </DropdownList.Item>
-        )}
+        {(playlist) => <PlayListChooserItem song={props.song} playlist={playlist} />}
       </For>
-    </DropdownList>
+    </div>
+  );
+};
+
+type PlayListChooserItemProps = {
+  song: Song;
+  playlist: PlaylistDropdown;
+};
+
+const PlayListChooserItem: Component<PlayListChooserItemProps> = (props) => {
+  return (
+    <DropdownList.Item
+      class="gap-3"
+      onClick={() => {
+        if (props.playlist.isOnSong) {
+          deleteSongFromPlaylist(props.playlist.name, props.song);
+        } else {
+          addToPlaylist(props.playlist.name, props.song);
+        }
+      }}
+    >
+      <div class="flex items-center gap-2">
+        <SongImage
+          class="size-7 flex-shrink-0 rounded bg-cover"
+          src={props.playlist.image}
```

Comment:
This only shows user uploaded images, it doesn't show the image of the first song in the playlist.